packages/reactive_ble_mobile/android/src/main/kotlin/com/signify/hue/flutterreactiveble/ble/ReactiveBleClient.kt

```diff
@@ -198,6 +198,33 @@ open class ReactiveBleClient(private val context: Context) : BleClient {
             RxBleConnection::writeCharWithoutResponse
         )
 
+    override fun writeDescriptorWithoutResponse(
+        deviceId: String,
+        serviceId: UUID,
+        characteristic: UUID,
+        descriptor: UUID,
+        value: ByteArray
+    ): Single<CharOperationResult> {
+        return getConnection(deviceId)
+            .flatMapSingle<CharOperationResult> { connectionResult ->
+                when (connectionResult) {
+                    is EstablishedConnection -> {
+                        connectionResult.rxConnection.writeDescriptor(serviceId, characteristic, descriptor, value)
+                            .andThen(Single.just(CharOperationSuccessful(deviceId, value.asList())))
+                    }
+                    is EstablishConnectionFailure -> {
+                        Single.just(
+                            CharOperationFailed(
+                                deviceId,
+                                "failed to connect ${connectionResult.errorMessage}"
+                            )
+                        )
+                    }
+                }
+            }.first(CharOperationFailed(deviceId, "Writechar timed-out"))
```

Comment:
@remonh87 `CharOperation*` types are used multiple types in this method (and possibly in other places), while it's not exactly about characteristics, but rather about descriptors (which, admittedly, are associated with characteristics). Do you think it's fine or do you think a new set of types should be introduced?

I don't have a strong opinion on this, mainly because I'm not that familiar with the Android part. Just drawing your attention.